src/api.js

```diff
@@ -1,42 +1,50 @@
 import axios from "axios";
 
-const Ncnews = axios.create({
+const ncNews = axios.create({
   baseURL: "https://project-x8zn.onrender.com/api/",
 });
 const apiUri = "https://project-x8zn.onrender.com/api/";
 
 export const getAllArticles = async () => {
-  const response = await Ncnews.get("articles");
+  const response = await ncNews.get("articles");
+  return response.data;
+};
+export const getTopics = async () => {
+  const response = await ncNews.get("topics");
+  return response.data;
+};
+export const getArticlesByQuery = async (topic) => {
+  const response = await ncNews.get(`articles?topic=${topic}`);
+
   return response.data;
 };
```

Comment:
Great that you are using your endpoint query to get all the articles from a certain topic 🙌 
Rather than make a separate function which gets the articles of a certain topic, it would be a good idea to make `getAllArticles` a bit more flexible, and optionally take in a topic parameter, and add the query if it doesn't.
The reason this would be good is it means you could use the same api function across multiple components, and could help negate the need for a separate `Topic` component.